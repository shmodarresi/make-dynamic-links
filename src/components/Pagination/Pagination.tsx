import { Button, Center, Flex, HStack, Stack } from "@chakra-ui/react";
import React from "react";

interface PaginationProps {
  noPages: number;
  currentPage: number;
  setCurrentPage: React.Dispatch<React.SetStateAction<number>>;
}

export const Pagination: React.FC<PaginationProps> = ({
  noPages,
  currentPage,
  setCurrentPage,
}) => {
  const pageNumbers = Array.from(Array(noPages + 1).keys()).slice(1);

  const nextPage = () => {
    if (currentPage !== noPages) setCurrentPage(currentPage + 1);
  };
  const prevPage = () => {
    if (currentPage !== 1) setCurrentPage(currentPage - 1);
  };
  return (
    <Center py={8}>
      <HStack gap={1}>
        <Button onClick={prevPage}>Prev</Button>
        {pageNumbers.map((pgNumber) => (
          <Button
            key={pgNumber}
            onClick={() => setCurrentPage(pgNumber)}
            colorScheme={currentPage === pgNumber ? "green" : undefined}
          >
            {pgNumber}
          </Button>
        ))}
        <Button onClick={nextPage}>Next</Button>
      </HStack>
    </Center>
  );
};
