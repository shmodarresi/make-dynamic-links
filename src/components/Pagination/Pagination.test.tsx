import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import { Pagination } from "./Pagination";

const mockSetState = jest.fn();

describe("Snapshot testing Pagination Component", () => {
  it("renders correctly", () => {
    const Component = renderer
      .create(
        <Pagination
          noPages={10}
          currentPage={5}
          setCurrentPage={mockSetState}
        />
      )
      .toJSON();
    expect(Component).toMatchSnapshot();
  });
});

describe("Element testing Pagination Component", () => {
  const currentPage = 5;
  it("render correctly elements", async () => {
    render(
      <Pagination
        noPages={10}
        currentPage={currentPage}
        setCurrentPage={mockSetState}
      />
    );
    const nodeElement = screen.getByText(/Prev/i);
    const nodeElements = screen.getAllByRole("button");

    expect(nodeElement).toBeInTheDocument();
    expect(nodeElements.length).toEqual(12);
    userEvent.click(screen.getByText("2"));
    expect(mockSetState).toHaveBeenCalledTimes(1);
  });
});
