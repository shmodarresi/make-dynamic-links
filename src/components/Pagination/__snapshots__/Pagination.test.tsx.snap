// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Pagination Component renders correctly 1`] = `
<div
  className="css-1l9nqp8"
>
  <div
    className="chakra-stack css-1a4j7nd"
  >
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      Prev
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      10
    </button>
    <button
      className="chakra-button css-4xx2wk"
      onClick={[Function]}
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
